src/Chapter2.hs

```diff
@@ -842,7 +867,9 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate n xs = if (n /= (abs n)) then
+  [] 
+  else take (length xs) $ drop n $ (cycle xs)
```

Comment:
Unfortunately, `cycle` fails in runtime on empty lists :recycle: So you need to handle this case separately for this function to work properly.

Reply:
Also, you can optimise the solution by dropping only `mod n (length of the list)` 🚤